server/controllers/dashboardController.js

```diff
@@ -1,27 +1,80 @@
 const logger = require('../../log');
+const audit = require('../data/audit');
+
+const {getOffenders} = require('../data/delius');
+const {getUpcomingReleases} = require('../data/nomis');
+const {getLicences} = require('../data/licences');
 
 exports.getIndex = function(req, res) {
-    logger.debug('GET /dashboard');
 
-    return res.render('dashboard/index', dashboardInfo);
+    const user = getLoggedInUserId();
+
+    audit.record('VIEW_DASHBOARD', user);
+
+    return getDashboardDetail(res, user);
 };
 
+function getLoggedInUserId() {
+    // todo
+    return '1';
+}
 
-const dashboardInfo = {
-    required: [
-        {
-            name: 'Andrews, Mark',
-            nomsId: 'A1235HG',
-            establishment: 'HMP Manchester',
-            dischargeDate: '01/11/2017',
-            inProgress: false
-        },
-        {
-            name: 'Bryanston, David',
-            nomsId: 'A6627JH',
-            establishment: 'HMP Birmingham',
-            dischargeDate: '10/07/2017',
-            inProgress: true
+async function getDashboardDetail(res, user) {
+
+    try {
+        const offenders = await getOffenders(user);
+
+        if (isEmpty(offenders)) {
+            return res.render('dashboard/index');
         }
-    ]
-};
+
+        const upcomingReleases = await getUpcomingReleases(offenders.nomisIds);
+
+        if (isEmpty(upcomingReleases)) {
+            return res.render('dashboard/index');
+        }
+
+        const activeLicences = await getLicences(getOffenderNomisIds(upcomingReleases));
+        const dashboardInfo = parseDashboardInfo(upcomingReleases, activeLicences);
+
+        return res.render('dashboard/index', dashboardInfo);
+
+
+    } catch (error) {
+        logger.error('Error during getDashboardDetail: ', error.message);
+        return renderErrorPage(res, error);
+    }
+}
+
+function isEmpty(candidateArray) {
+    return !candidateArray || candidateArray.length <= 0;
+}
+
+function getOffenderNomisIds(releases) {
+    return releases.map(offender => offender.nomisId);
+}
+
+function parseDashboardInfo(upcomingReleases, activeLicences) {
+
+    const required = upcomingReleases.map(offender => {
+        const licence = activeLicences.find(licence => licence.nomisId === offender.nomisId);
+        return licence ? Object.assign(offender, {inProgress: true, licenceId: licence.id}) : offender;
+    });
```

Comment:
With this version of node we can use the spread operator to merge I believe. 

Bit less verbose: {...obj1, ...obj2}